### PR TITLE
Options cleanup and documentation: Wide review welcome!

### PR DESCRIPTION
**Pull request by rustyrussell**
Nobody reads it, but sometimes you have to write it anyway.

### CODE REVIEW

File: doc/lightningd-config.5.txt [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    can call their node anything they want.  The default is an
+    NSA-style codename derived from your public key, but "Peter Todd"
+    and "VAULTERO" are good options, too.
+
+*fee-base*='MILLISATOSHI'::
+    The base fee to charge for every payment which passes through.  Note that
+    millisatoshis are a very, very small unit!
+
+*fee-per-satoshi*='MILLIONTHS'::
+    This is the proportional fee to charge for every payment which passes
```
Comment (ajtowns) [inline doc/lightningd-config.5.txt]: `fee-base` and `fee-per-satoshi` would be better as something like `fee-per-transaction` and `fee-per-value` or similar, wouldn't they? `millisatoshis-per-kilosatoshi` would be a more accurate, but probably not helpful, description of the value of the latter field. I wonder if `fee-percent` and saying its denominated in 10,000ths of a percent mightn't be clearer though?

File: doc/lightningd-config.5.txt [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+*ignore-fee-limits*='BOOL'::
+    Allow nodes which establish channels to us to set any fee they
+    want.  This may result in a channel which cannot be closed, should
+    fees increase, but make channels far more reliable since we never
+    close it due to unreasonable fees.
+
+*commit-time*='TIME'::
+    How long to wait before sending commitment messages to the peer: in
+    theory increasing this would reduce load, but your node would have to be
+    extremely busy node for you to even notice.
```
Comment (ajtowns) [inline doc/lightningd-config.5.txt]: This doesn't say what units are used -- I guess seconds, but milliseconds would probably be reasonable too?

File: doc/lightningd-config.5.txt [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+Lightning channel and HTLC options:
+
+*locktime-blocks*='BLOCKS'::
+    How long we need to spot an outdated close attempt, which is also
+    how long the peer would need to wait if they perform a unilateral
+    close.
+
+*max-locktime-blocks*='BLOCKS'::
```
Comment (ajtowns) [inline doc/lightningd-config.5.txt]: It seems confusing that "locktime-blocks" is ours, while "max-locktime-blocks" is theirs.

File: doc/lightningd-config.5.txt [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    case.  If they ask for longer, we'll refuse to create a channel,
+    and if an HTLC asks for longer, we'll refuse it.
+
+*funding-confirms*='BLOCKS'::
+    Confirmations required for the funding transaction when the other side
+    opens a channel before the channel is usable.
+
+*commit-fee-min*='PERCENT'::
+*commit-fee-max*='PERCENT'::
+    Limits on what onchain fee range we'll allow when a node opens a
```
Comment (ajtowns) [inline doc/lightningd-config.5.txt]: Is this also a percentage of `estimatesmartfee 2`? (Shouldn't the "2" be configurable?)

File: doc/lightningd-config.5.txt (hunk 1 above)
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: All names have issues, AFAICT.   The spec calls them `fee_base_msat` and `fee_proportional_millionths`.

File: doc/lightningd-config.5.txt (hunk 2 above)
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: It's actually quite a comprehensive parser that allows 'ns/nanosecond/nanoseconds' through to 'd/day/days'.  Which is complete overkill, and only used in two places.  Will fix.

File: doc/lightningd-config.5.txt (hunk 4 above)
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: '2' is the smallest you can ask bitcoind for, ie: I really want this now.

File: doc/lightningd-config.5.txt (hunk 3 above)
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: Not sure what names would be clearer though...

File: doc/lightningd-config.5.txt [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+'bind-addr' and 'announce-addr' options.
+
+*addr*='[IPADDRESS[:PORT]]|autotor:TORIPADDRESS[:TORPORT]'::
+
+    Set an IP address (v4 or v6) or automatic Tor address to listen on
+    and (maybe) announce as our node address.
+
+    An empty 'IPADDRESS' is a special value meaning bind to IPv4 and/or
+    IPv6 on all interfaces, '0.0.0.0' means bind to all IPv4
+    interfaces, '::' means 'bind to all IPv6 interfaces'.  'PORT' is
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: Thanks, fixed!

File: doc/lightningd-config.5.txt [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+    not specified, 9735 is used.
+
+    This option can be used multiple times to add more addresses, and
+    its use disables autolisten.  If necessary, and 'always-use-proxy'
+    is not specified, a DNS lookup may be done to resolve 'IPADDRESS'.
+
+*announce-addr*='[IPADDRESS[:PORT]]|TORADDRESS.onion[:PORT]'::
+
+    Set an IP address or Tor address to announce; a Tor address is
+    distinguished by ending in '.onion', and is   'PORT' defaults to 9735.
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: Yes, deleted ', and is' and replaced with '.'.

File: doc/lightningd-config.5.txt [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+
+    Empty or wildcard IPv4 and IPv6 addresses don't make sense here.
+    Also, unlike the 'addr' option, there is no checking that your
+    announced addresses are sane (ie. not localhost).
+
+    This option can be used multiple times to add more addresses, and
+    its use disables autolisten.  If necessary, and 'always-use-proxy'
+    is not specified, a DNS lookup may be done to resolve 'IPADDRESS'
+    or 'TORADDRESS'.
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: Agreed., fixed.

File: doc/lightningd-config.5.txt [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+    An empty 'IPADDRESS' is a special value meaning bind to IPv4 and/or
+    IPv6 on all interfaces, '0.0.0.0' means bind to all IPv4
+    interfaces, '::' means 'bind to all IPv6 interfaces'.  'PORT' is
+    not specified, 9735 is used.
+
+    This option can be used multiple times to add more addresses, and
+    its use disables autolisten.  If necessary, and 'always-use-proxy'
+    is not specified, a DNS lookup may be done to resolve 'IPADDRESS'.
+
+*announce-addr*='[IPADDRESS[:PORT]]|TORADDRESS.onion[:PORT]'::
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: Indeed, that is a cut&paste bug from the addr line, fixed.

File: doc/lightningd-config.5.txt (hunk 3 above)
Comment (ZmnSCPxj) [reply doc/lightningd-config.5.txt]: `our-locktime-blocks` `their-max-locktime-blocks`?

Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: But it's also used for htlc max. It really is the worst case lockup time, in blocks.

Comment (ZmnSCPxj) [reply doc/lightningd-config.5.txt]: So why is `our-locktime-blocks` and `their-max-locktime-blocks` not also appropriate for HTLC max, given the current name is just `locktime-blocks` and `max-locktime-blocks`?

Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: Perhaps max-locktime-blocks and watchtime-blocks? Want to emphasize the tradeoffs...

Comment (rustyrussell) [reply doc/lightningd-config.5.txt]: OK, I renamed locktime-blocks to watchtime-blocks, and reworked the descriptions a little:

```asciidoc
*watchtime-blocks*='BLOCKS'::
    How long we need to spot an outdated close attempt: on opening a channel
    we tell our peer that this is how long they'll have to wait this many
    blocks if they perform a unilateral close.

*max-locktime-blocks*='BLOCKS'::
    The longest our funds can be delayed (ie. the longest *watchtime-blocks*
    our peer can ask for, and also the longest HTLC timeout we will accept).
    If our peer asks for longer, we'll refuse to create a channel, and if an
    HTLC asks for longer, we'll refuse it.
```

**Comment by rustyrussell** [comment]
Rebased, test breakage fixed, and the uninitialized variable introduced by the original 'fix default feerate' patch fixed.

**Pull request merged**